### ISSUE TITLE
Model size and accuracy remains the same after compressing.

### ISSUE THREAD

**Issue by jimmystereos** [openvinotoolkit/nncf#959]
I'm facing two problem.
1. The size of the model remains the same after compressing whether I did or didn't do fine tuning. I'm using `torchinfo.summary` to identify the size.

2. The accuracy remains the same after compressing but before fine tuning. So I guess it need to do fine tuning to let the compression take effect. Is there any suggestion about this?

Thanks!

**Comment by vshampor** [comment]
Greetings, @jimmystereos!

1. The *torch representation* of an NNCF-compressed model will not have any size or complexity reductions when compared with a *torch representation* of the original model since at the training framework level, NNCF only _simulates_ the upcoming compression. The actual size reductions will only be visible at the OpenVINO Intermediate Representation level of the compressed model.

2. To clarify - if you obtained an accuracy for a model object leaving `create_compressed_model` that is equal to the original model accuracy even without finetuning, then congratulations, you have achieved acceleration with zero accuracy drop! It is not usually the case, however, and right after compression the accuracy goes down quite a bit, so fine-tuning is indeed in general required to recover the accuracy. 
The fine-tuning is also generally required for sparsity and pruning algos since the level of sparsity/pruning is being increased gradually along with the weight fine-tuning based on the selected sparsity/pruning level scheduler in the NNCF .json config file.

**Comment by jimmystereos** [comment]
Hi, thanks for your reply!
For the first question, I'd like to clarify that is this the case for both Tensorflow and Pytorch? 
We would not be able to use this tool if not using OpenVINO?

And fine-tuning is _"required"_ for sparsity and pruning algorithms (RB, Filter, Magnitude)?

These questions bother me a lot. Thanks for replying and providing great tools.

**Comment by vshampor** [comment]
The extent to which the NNCF-exported compressed models are usable outside OpenVINO are quite limited. You may try using the NNCF TF exported frozen graph .pb files in the regular TF runtimes and they *might* work. The NNCF PT exported ONNX files won't work in the quantization case with the regular `onnxruntime` unless you specify `"export_to_onnx_standard_ops": true` in the NNCF .json config and use an older `onnxruntime` version. The current positioning of NNCF is such that we really target OpenVINO as the inference framework for the NNCF-created models and do not guarantee functionality for the other inferencing frameworks, which is unfortunate, but is not something we can improve in the nearest future.

Fine-tuning is *generally* required for sparsity and pruning since sparsity and pruning is usually being introduced gradually during the course of the model fine-tuning. Of course, there may be such a miraculous situation where you would have a baseline model where the application of the target, ultimate sparsity/pruning level (80%, for instance) without further fine-tuning produces a model with a satisfactory accuracy drop, but this is not usually the case - the abrupt application of pruning and sparsity usually destroys the quality of the resulting model.
